lib/ae_mdw/node/db.ex

```diff
@@ -242,6 +242,27 @@ defmodule AeMdw.Node.Db do
     vsn
   end
 
+  @spec nonce_at_block(Blocks.block_hash(), pubkey()) :: non_neg_integer()
+  def nonce_at_block(mb_hash, account_pk) do
+    {:value, micro_block} = :aec_db.find_block(mb_hash)
+    header = :aec_blocks.to_header(micro_block)
+    {:ok, hash} = :aec_headers.hash_header(header)
+    consensus = :aec_headers.consensus_module(header)
+    node = {:node, header, hash, :micro}
+    prev_hash = :aec_block_insertion.node_prev_hash(node)
+
+    {:value, trees_in, _tree, _difficulty, _fees, _fraud} =
+      :aec_db.find_block_state_and_data(prev_hash, true)
```

Comment:
Regarding the `prev_hash` I understand that it shall use the `nonce` before it gets incremented by all transactions of the block but it seems the opposite should be handled as well when there were already other transactions by `account_pk` before the `Oracle.query`.

Reply:
Could it handle a `nonce` initialization considering the txs from the same sender that have already happened in the block?